tests/sentry/workflow_engine/processors/test_workflow.py

```diff
@@ -400,3 +408,75 @@ def test_enqueue_workflow__adds_to_workflow_engine_set(
                 {"event_id": self.event.event_id, "occurrence_id": self.group_event.occurrence_id}
             ),
         )
+
+
+class TestDeleteWorkflow(BaseWorkflowTest):
+    def setUp(self):
+        self.workflow = self.create_workflow()
+        workflow_trigger = self.create_data_condition_group()
+        self.workflow.when_condition_group = workflow_trigger
+        self.workflow.save()
+
+        action_filter = self.create_data_condition_group()
+        action = self.create_action()
+        action_and_filter = self.create_data_condition_group_action(
+            condition_group=action_filter,
+            action=action,
+        )
+
+        workflow_actions = self.create_workflow_data_condition_group(
+            workflow=self.workflow, condition_group=action_filter
+        )
+
+        trigger_condition = self.create_data_condition(condition_group=workflow_trigger)
+        action_condition = self.create_data_condition(condition_group=action_filter)
+
+        self.workflow_id = self.workflow.id
+        self.workflow_trigger_id = workflow_trigger.id
+        self.action_filter_id = action_filter.id
+        self.action_id = action.id
+        self.action_and_filter_id = action_and_filter.id
+        self.workflow_actions_id = workflow_actions.id
+        self.trigger_condition_id = trigger_condition.id
+        self.action_condition_id = action_condition.id
+
+    # TODO - Figure out why parametrize isn't working for `test_delete_workflow`
+    def assert_clean_workflow_models(self):
+        assert not Workflow.objects.filter(id=self.workflow_id).exists()
+        assert not DataConditionGroup.objects.filter(id=self.workflow_trigger_id).exists()
+        assert not DataCondition.objects.filter(id=self.trigger_condition_id).exists()
+
+    def assert_clean_action_models(self):
+        assert not DataConditionGroup.objects.filter(id=self.action_filter_id).exists()
+        assert not Action.objects.filter(id=self.action_id).exists()
+        assert not DataConditionGroupAction.objects.filter(id=self.action_and_filter_id).exists()
+        assert not WorkflowDataConditionGroup.objects.filter(id=self.workflow_actions_id).exists()
+        assert not DataCondition.objects.filter(id=self.action_condition_id).exists()
+
+    def test_delete_workflow__deletes_workflow(self):
+        delete_workflow(self.workflow)
+        self.assert_clean_workflow_models()
+
+    def test_delete_workflow__deletes_action_tables(self):
+        delete_workflow(self.workflow)
+        self.assert_clean_action_models()
```

Comment:
I feel like ideally, this would be a parameterized test that i could ensure each of these models are cleaned up individually. that said, i couldn't get [`@pytest.mark.parametrize`](https://docs.pytest.org/en/stable/example/parametrize.html) to successfully pass in a param dynamically here. Any one know how get that working? 🤔  (i couldn't get it to work to pass in a static string 😢 and copy pasta from the docs)

Reply:
Looks like the reason is because it's base class is extending from `TestCase`: https://github.com/pytest-dev/pytest/issues/1271 😢